src/component.js

```diff
@@ -88,6 +88,9 @@ export default class Component {
                 // Don't react to data changes for cases like the `x-created` hook.
                 if (self.pauseReactivity) return
 
+                // Don't react if the mutated key is array length as it is a consequence of another mutated property like pushing to the array
+                if (Array.isArray(target) && key === 'length' ) return
```

Comment:
Are we sure we want to do the `Array.isArray` check?

What about things like strings?

Reply:
`Array.isArray` is not that necessary but I think it is better to include it for two reasons:

1. To be self-explanatory about what we mean by length property. 
2. This fix is to reverse observable-membrane unwanted behavior which verifies first if `(key === 'length' && isArray(originalTarget))` So it is better to narrow down the fix to only this scenario.

Regarding strings will work with or without this fix as observable-membrane specifies this unwanted behavior to array length mutations.

Another side note, If the component x-data object contains a length property will also work as `(oldValue !== value)` will be true.

Reply:
Makes sense to me 👍

Reply:
👍 Strings are immutable object in js as well (string.length is readonly so it will never mutate).

Have you tried if other methods such as array.pop and array.shift are still triggering a refresh?

Reply:
You're right, I was thinking about maybe Set/Map but that stores the number of items under `.size`

I think that's beyond the scope of this PR though, should be a follow up PR

Reply:
array.pop, shift, push, unshift are fine as the change in index length will be ignored by the fix. While the changes in the indexes by shift and unshift will cause repeated rendering for each single index change but ar lease one less rendering related to index length. The same thing for or reverse, splice, fill, ...

Reply:
Cool, just missing tests to avoid this fix regressing then

Reply:
Will do right now